spacy/errors.py

```diff
@@ -358,6 +358,11 @@ class Errors(object):
             "arguments to exclude fields from being serialized or deserialized "
             "is now deprecated. Please use the `exclude` argument instead. "
             "For example: exclude=['{arg}'].")
+    E129 = ("Can not write the label of an existing Span object because a Span "
+            "is only a read-only view of the underlying Token objects stored in the Doc. "
+            "Instead, create a new Span object and specify the `label` keyword argument, "
+            "for example:\nfrom spacy.tokens import Span\n"
+            "span = Span(doc, start=3, end=5, label='GPE')")
```

Comment:
If we wanted to, we could be *super* helpful here and pass in the `start`, `end` and `label` from the span / label the user was trying to add. So they'd be able to just copy-paste the code 😃

(Or is this overkill? Any potential bad side-effects of this that I'm missing? cc: @honnibal)

Reply:
Good idea, it can hardly be overkill if it's only a minute's work :-)

Reply:
The only edge case maybe that isn't covered now is passing "None", e.g.
`span.label_ = None`
will print, at the end of the error msg:
`span = Span(doc, start=0, end=1, label='None')`

Reply:
This is a great error message! Tonnes better than my suggestion of just removing the setter...